zig/private/bzlmod/zig.bzl

```diff
@@ -73,24 +94,85 @@ def handle_tags(module_ctx, *, known_versions):
 
     return None, versions
 
-def _parse_zig_versions_json(json_string):
+def parse_zig_versions_json(json_string):
+    """Parse a Zig SDK versions index in JSON format.
+
+    Exposed as a standalone function for unit testing.
+
+    Args:
+      json_string: String, The version index in JSON format.
+
+    Returns:
+      (err, data), maybe an error or a
+        `dict[version, dict[platform, struct(url, sha256)]]`.
+    """
     result = {}
 
-    data = json.decode(json_string)
+    data = json.decode(json_string, default = None)
+
+    if data == None:
+        return "Invalid JSON format in Zig SDK version index.", None
+
     for version, platforms in data.items():
+        if "version" in platforms:
+            version = platforms["version"]
+
+        if not semver.is_valid(version):
+            return "Malformed version number '{}' in Zig SDK version index.".format(version), None
+
         for platform, info in platforms.items():
+            if type(info) != "dict" or not platform in PLATFORMS:
```

Comment:
I don't think this can ever be true: type() returns a class, never a string.

Reply:
This is Starlark, not Python. type returns a string in this case https://bazel.build/rules/lib/globals/all#type